### DISCUSSION TITLE
helm -  installation deploy stuck in "pending" state

### DISCUSSION THREAD

**Discussion by roberto-maggi** [kimai/kimai#4289] [On-Premise: Installation & Updates]
Hi you all,
I'm new to Kimai and I'm working on a staging installation for my company.
In installed the sw via helm with :
helm install my-kimai2 robjuz/kimai2 --version 3.1.2
as suggested but the deploy is stuck in pending state, where a "kubectl describe po kimai-kimai2-7d595cd49c-fr7tf" results in a 

root@k-master-1:~# k describe po kimai-kimai2-7d595cd49c-fr7tf
Name:             kimai-kimai2-7d595cd49c-fr7tf
Namespace:        default
Priority:         0
Service Account:  default
Node:             <none>
Labels:           app.kubernetes.io/instance=kimai
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=kimai2
                  helm.sh/chart=kimai2-3.1.2
                  pod-template-hash=7d595cd49c
Annotations:      checksum/config: f831063caa5b466866f84535573de0b768cea918e99257818702be04b616421b
                  checksum/secret: e91717e28f6af030ddb90eb719a4cb8d451ec5ba703c3aec20a8b9f413047930
Status:           Pending
IP:               
IPs:              <none>
Controlled By:    ReplicaSet/kimai-kimai2-7d595cd49c
Containers:
  kimai2:
    Image:      docker.io/kimai/kimai2:apache-2.0.23-prod
    Port:       8001/TCP
    Host Port:  0/TCP
    Limits:
      memory:  256Mi
    Requests:
      cpu:      100m
      memory:   256Mi
    Liveness:   http-get http://:http/en/login delay=120s timeout=5s period=10s #success=1 #failure=6
    Readiness:  http-get http://:http/en/login delay=30s timeout=5s period=10s #success=1 #failure=6
    Environment:
      APP_ENV:        prod
      APP_SECRET:     <set to the key 'app-secret' in secret 'kimai-kimai2'>  Optional: false
      ADMINMAIL:      
      ADMINPASS:      <set to the key 'admin-password' in secret 'kimai-kimai2'>  Optional: false
      MAILER_FROM:    kimai@example.com
      MAILER_URL:     <set to the key 'mailer-url' in secret 'kimai-kimai2'>    Optional: false
      DATABASE_URL:   <set to the key 'database-url' in secret 'kimai-kimai2'>  Optional: false
      TRUSTED_HOSTS:  localhost,kimai.local
    Mounts:
      /opt/kimai/config/packages/local.yaml from config (rw,path="local.yaml")
      /opt/kimai/var from kimai-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-k6k9j (ro)
Conditions:
  Type           Status
  PodScheduled   False 
Volumes:
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      kimai-kimai2-config
    Optional:  false
  kimai-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  kimai-kimai2
    ReadOnly:   false
  kube-api-access-k6k9j:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason            Age                  From               Message
  ----     ------            ----                 ----               -------
  Warning  FailedScheduling  3m17s (x5 over 23m)  default-scheduler  0/3 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/3 nodes are available: 3 No preemption victims found for incoming pod..

it sounds like the nodes are too small in resources but the whole cluster is brand new, created just for this

root@k-master-1:~# for MACHINE in k-master-1 k-worker-1 k-worker-2 ; do ssh $MACHINE hostname && df -h | grep ubuntu--vg-ubuntu--lv && free -h  && nproc ; done
k-master-1
/dev/mapper/ubuntu--vg-ubuntu--lv   12G  8.7G  2.0G  82% /
              total        used        free      shared  buff/cache   available
Mem:          3.8Gi       970Mi       666Mi        16Mi       2.2Gi       2.6Gi
Swap:            0B          0B          0B
2
k-worker-1
/dev/mapper/ubuntu--vg-ubuntu--lv   12G  8.7G  2.0G  82% /
              total        used        free      shared  buff/cache   available
Mem:          3.8Gi       970Mi       665Mi        16Mi       2.2Gi       2.6Gi
Swap:            0B          0B          0B
2
k-worker-2
/dev/mapper/ubuntu--vg-ubuntu--lv   12G  8.7G  2.0G  82% /
              total        used        free      shared  buff/cache   available
Mem:          3.8Gi       971Mi       665Mi        16Mi       2.2Gi       2.6Gi
Swap:            0B          0B          0B
2

I already created a free persistent volume but can't guess what I'm missing.
Any help will be appreciated, thanks in advance

**Comment by kevinpapst** [comment]
Wrong repo, go here: https://github.com/robjuz/helm-charts

**Reply by roberto-maggi** [reply]
thanks for the prompt reply.

root@k-master-1:~# helm repo list
NAME         	URL                                       
robjuz       	https://robjuz.github.io/helm-charts      

I did it but had the same issue

Events:
  Type     Reason            Age                From               Message
  ----     ------            ----               ----               -------
  Warning  FailedScheduling  49s (x2 over 50s)  default-scheduler  0/3 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/3 nodes are available: 3 No preemption victims found for incoming pod..

**Reply by kevinpapst** [reply]
I have no clue what you want to tell me, but I do NOT care about the containerization or how you use my software.

You have to ask the creator of the Helm chart.

**Reply by roberto-maggi** [reply]
ok thanks again
